### DISCUSSION TITLE
Enquiring About Adaptive AUTOSAR: Application Demos and Understanding ARXML

### DISCUSSION THREAD

**Discussion by KEDW0316** [langroodi/Adaptive-AUTOSAR#31] [Q&A] [answered]
Hello, I am a master's student researcher starting to study Adaptive AUTOSAR.

Firstly, I would like to express my gratitude for this project. It has allowed me to gain a deeper understanding of Adaptive AUTOSAR, which I had previously only studied theoretically.

However, as I try to execute the project and understand it more deeply, I realize that my knowledge is limited and I have some questions. Before I ask them, I would like to apologize in advance if my questions seem trivial or obvious.

Is there an application demo using the Adaptive AUTOSAR stack here? I heard it was supposed to be released in May 2023, but I couldn't find it in the process of building and running. I'm not sure if I just missed it. Is there anything I can refer to for developing an application on the Adaptive AUTOSAR simulator?

I'm not quite sure about the exact role of ARXML. In my personal studies related to software development based on Adaptive AUTOSAR, it was explained that one uses an Adaptive AUTOSAR tool provided by a specific company to configure ARXML, and then this file is converted to a Json Manifest file and Codes. In this simulator, is ARXML used in a way to define arguments like IP?

Thank you for reading this.

Have a nice day!

**Comment by langroodi** [comment, marked as answer]
Hi there,

Happy to hear that the project became useful in your study/research.

Regarding your first question, I would guess that you might miss it. You can watch the demo video from here:
https://youtu.be/OFYA-raVrWE?t=2590

And yes, it has a demo application which its source is accessible from here:
https://github.com/langroodi/Adaptive-AUTOSAR/tree/master/src/application

And about the second question, you are right about ARXML config tools. I did not use those tools because:
- The famous ones there in the market are commercial and closed source;
- This simulator is not fully compatible with all the ARXML schema and it supports a tiny part of that.
So, the ARXML file has been created manually and it is fed into the simulator directly as the XML and not any other format such as JSON. The reason behind that was only to show the idea of how to config the ARA platform using the config file. And yes, the ARXML here also contains IP layer information.

**Reply by KEDW0316** [reply]
Thank you so much!

**Reply by KEDW0316** [reply]
I have an additional question to confirm my understanding. Below is what I have written based on my understanding of the project and applications of Adaptive AUTOSAR.

This project is set up to implement and simulate some of the Adaptive AUTOSAR standards. This is implemented in the ara folder. Ara is the runtime for Adaptive AUTOSAR, and using the implemented ara, various applications can be developed in accordance with Adaptive AUTOSAR standards.

Already implemented application demos include applications like diagnostic_manager and platform_health_management. Each of these uses various software provided by ara to offer new services. These implemented applications comply with the standards of Adaptive AUTOSAR.

Thank you for reading and taking the time to respond.

Have a great day!

**Reply by langroodi** [reply]
The both statements you made are totally correct. I only need to add something regarding your second statement:
Applications within AUTOSAR Adaptive Platform (AP) can be categorized into:
1. Platform Applications (Functional Clusters);
2. Adaptive Applications (AA).

Functional Clusters are default applications defined by the standard that most of them are required to make the AP up and running. The `diagnostic_manager` (DM) and `platform_health_management` (PHM) which you mentioned are both functional clusters. In this project they ended up in `application/platform` directory.

However, Adaptive Applications are manufacture specific instances and they are **not required** by the standard and they just extend the functionality of the platform. The `extended_vehicle` is an example of that. In this project the AA ended up inside the `application`, but outside of the `platform` directory.

For more information: https://www.autosar.org/fileadmin/standards/R22-11/AP/AUTOSAR_TR_FunctionalClusterShortnames.pdf

**Reply by KEDW0316** [reply]
Thank you so much!
It was very helpful!
